packages/payment/src/StripeService.ts

```diff
@@ -0,0 +1,101 @@
+import { injectable } from 'inversify';
+import { WishCardRepository } from '@donategifts/wishcard';
+import { UserRepository } from '@donategifts/user-data';
+import { logger } from '@donategifts/helper';
+import Stripe from 'stripe';
+import { IDonationHook, IStripeIntent, ObjectId } from '@donategifts/common';
+import { calculateWishItemTotalPrice, handleDonation } from './helper/paymentHelper';
+
+// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+const stripe = new Stripe(process.env.STRIPE_KEY!, {
+	apiVersion: '2020-08-27',
+});
+
+@injectable()
+export class StripeService {
+	constructor(
+		private wishCardRepository: typeof WishCardRepository = WishCardRepository,
+		private lastWishcardDonation: string = '',
+		private userRepository: typeof UserRepository = UserRepository,
+	) {}
+
+	public async createIntent(
+		stripeData: IStripeIntent,
+	): Promise<{ success: boolean; error?: string; clientSecret?: string }> {
+		const { wishCardId, email, agencyName, userDonation, userId } = stripeData;
+		// Create a PaymentIntent with the order amount and currency
+		const wishCard = await this.wishCardRepository.getWishCardByObjectId(ObjectId(wishCardId));
+
+		const user = await this.userRepository.getUserById(ObjectId(userId));
+
+		// By default stripe accepts "pennies" and we are storing in a full "dollars". 1$ == 100
+		// so we need to multiple our price by 100. Genious explanation
+		const PENNY = 100;
+		let totalItemPrice = await calculateWishItemTotalPrice(wishCard.wishItemPrice);
+		if (!wishCard) {

```

Comment:
move this up right after the wishcard query so the user doesn't get executed if there's no wishcard :D